src/EnergyPlus/ZoneTempPredictorCorrector.cc

```diff
@@ -714,9 +714,20 @@ void GetZoneAirSetPoints(EnergyPlusData &state)
             CTSchedMapToControlledZone(TempControlledZoneNum) = tempZone.setptTypeSched->Num;
         }
 
-        for (HVAC::SetptType setptType : HVAC::setptTypes) {
+        for (HVAC::SetptType setptType : HVAC::controlledSetptTypes) {
             auto const &setpt = tempZone.setpts[(int)setptType];
+
             if (!setpt.isUsed) {
+                // Catch early issues
+                if (tempZone.setptTypeSched->hasVal(state, (int)setptType)) {
+                    ShowSevereError(state, format("Control Type Schedule={}", tempZone.setptTypeSched->Name));
+                    ShowContinueError(
+                        state,
+                        format("..specifies {} ({}) as the control type. Not valid for this zone.", (int)setptType, setptTypeNames[(int)setptType]));
+                    ShowContinueError(state, format("..reference {}={}", cZControlTypes((int)ZoneControlTypes::TStat), tempZone.Name));
+                    ShowContinueError(state, format("..reference ZONE={}", tempZone.ZoneName));
+                    ErrorsFound = true;
+                }
```

Comment:
I'm trying to think about what happens if someone uses an API actuator to adjust the schedule value later.  I know there are probably lots of places in the code that this could cause some issues, so not necessarily saying we need to do anything here, just thinking about possible impacts.